app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -320,3 +383,66 @@ const intoConfirmSignUpErrorOrThrow = (error: AmplifyError): ConfirmSignUpError
 
     throw error
 }
+
+// ========================
+// === SignInWithGoogle ===
+// ========================
+
+const signInWithGoogle = async (customState?: string) =>
```

Comment:
1. as agreed in other PR, lets use `function`s instead of top-level const lambdas.
2. You can refactor `const provider = amplify.CognitoHostedUIIdentityProvider.Google` to make the body contain more one line expressions.

Reply:
1. all lambdas should now be functions (thanks to eslint lints)
2. still to be done

Reply:
2. Fixed.